com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKPokeReticleVisual.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    public class MRTKPokeReticleVisual : BaseReticleVisual
+    {
+        [SerializeField]
+        [Tooltip("The interactor which this visual represents.")]
+        private PokeInteractor pokeInteractor;
+
+        [SerializeField]
+        [Tooltip("The GameObject which holds the proximity light for the reticle")]
+        private GameObject proximityLight;
+
+        protected void OnEnable()
+        {
+            Application.onBeforeRender += UpdateReticle;
+        }
+
+        protected void OnDisable()
+        {
+            UpdateReticle();
+            Application.onBeforeRender -= UpdateReticle;
+        }
+
+        private static readonly ProfilerMarker UpdateVisualsPerfMarker =
+            new ProfilerMarker("[MRTK] MRTKPokeReticleVisual.UpdateReticle");
+
+        [BeforeRenderOrder(XRInteractionUpdateOrder.k_BeforeRenderLineVisual)]
+        private void UpdateReticle()
+        {
+            using (UpdateVisualsPerfMarker.Auto())
+            {
+                reticle.SetActive(pokeInteractor.enabled && pokeInteractor.isHoverActive);
+
+                // TODO: Ideally we'd want the ReticleVisualScript to be responsible for all aspects of a visuals appearance.
+                // This is because it would allow us to freely use different reticle icons without changing other behavior.
+                // However, until the ReticleMagnetism class is revisited and potentially incorporated into this class, we'll
+                // leave all pose related responsibilities to the ReticleMagnetism component.
+                // reticle.transform.SetPositionAndRotation(pokeInteractor.PokeTrajectory.End, pokeInteractor.attachTransform.rotation);
+
+                // The proximity light should only be active when the reticle is
+                if (proximityLight.gameObject != null)
+                {
+                    proximityLight.SetActive(reticle.activeSelf);
```

Comment:
What was this handled by previously (if anything)?

Reply:
This was handled implicitly because we used to disable the whole gameobject through the interactor itself. Now, the reticle is in charge of it's own behavior and manually needs to disable the reticle/proximity light which are parented to it.

Reply:
Why is the proximity light not just a child of the reticle and enabled/disabled along with the reticle?

Reply:
The structure of the interactor visual is

Reticle Visual (the script we have that manages everything)
    Reticle Model (gets switched out as part of the `IXRCustomReticleProvider` pattern)
    Prox Light 
    
 We want to keep the proximity light separate because its something which is pretty important for making XR interactions feel more immersive. The proximity light on the poke interactor helps you recognize that you are close to an object for example. If the light was attached to the reticle model, the light can now easily be removed/replaced unwittingly by a mis-configured custom reticle. I think conceptually, its best to think of the Reticle Model as **just** a model as much as possible. We can easily change the pattern back if we find that there are many cases where having no proximity light is desireable.